src/components/Data/Layout.vue

```diff
@@ -19,27 +19,120 @@
 <script>
 import MainSpinner from '../Common/MainSpinner'
 import { Multipane, MultipaneResizer } from 'vue-multipane'
+import Treeview from '@/components/Data/Leftnav/Treeview'
 
 export default {
   name: 'DataLayout',
   components: {
     MainSpinner,
-    // Treeview,
+    Treeview,
     Multipane,
     MultipaneResizer
   },
+  data() {
+    return {
+      isFetching: false
+    }
+  },
   computed: {
     loading() {
-      return this.$store.direct.state.index.loadingIndexes
+      return (
+        this.isFetching ||
```

Comment:
I wonder if this flag is really necessary

Reply:
in case the loading state has not yet started (it depends on which action is performed first) and the child components try to render while the data is not yet present